test/pylib/cpp/facade.py

```diff
@@ -65,17 +67,25 @@ def run_test(self, executable: Path, original_name: str, test_id: str, mode: str
          raise NotImplementedError
 
 
-def run_process(args: list[str], timeout):
+def run_process(args: list[str], timeout, env: dict = None) -> tuple[subprocess.Popen[str], str]:
     args = shlex.split(' '.join(args))
-    p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    if env:
+        env.update(os.environ)
```

Comment:
What if env is none or empty? In that case you don't update it with os.environ and pass it as is to subprocess.Popen. As a result, the subprocess is run with blank environment? Or does it "falls back" to os.environ under the hood?